### DISCUSSION TITLE
Lower & Upper Bound

### DISCUSSION THREAD

**Discussion by ayushpatidar** [ourownstory/neural_prophet#506] [Q&A - get help using NeuralProphet]
How can we get lower & upper bound in NeuralProphet.  

Below is the code that i am using: 

nprophet_model = NeuralProphet()
metrics = nprophet_model.fit(df_train, freq="1H")
future_df = nprophet_model.make_future_dataframe(df_train, 
                                                 periods = test_length, 
                                                 n_historic_predictions=len(df_train))
preds_df_2 = nprophet_model.predict(future_df)

**Comment by Carminho1** [comment]
Currently, I am facing the same issue. I am working on a model for photovolatic-power-production and obviously the photovolatic-power-production has to be always equal to zero at night (which is also always the case in the data to fit my model). Unfortunately, my forecast predictions somehow take values of lower than zero during nighttime periods.
![Bildschirmfoto 2022-01-04 um 20 03 02](https://user-images.githubusercontent.com/95876970/148110310-dfd32087-f651-4980-8df4-a3bea8535fc6.jpg)

It would be great if somebody knows how to implement a lower bound to my photovolatic-power-production forecast, so that the forecast will never be lower than 0.

Many Thanks in Advance.
